src/MinkowskiSum.jl

```diff
@@ -635,10 +639,9 @@ end
 # ================
 
 @inline function σ_helper(d::AbstractVector{N},
-                          array::AbstractVector{<:LazySet}
-                         )::Vector{N} where {N<:Real}
-    svec = zeros(N, length(d))
-    for sj in array
+                          array::AbstractVector{<:LazySet}) where {N<:Real}
+    svec = similar(d)
```

Comment:
You have to start from a zero vector. So I would just keep it as before.

Reply:
You are right, `similar` is wrong here. There is no generic `zeros` for an abstract vector. I wanted to avoid converting to a dense vector, because `d` is of abstract type. I'll try with [mapreduce](https://docs.julialang.org/en/v1/base/collections/#Base.mapreduce-Tuple{Any,Any,Any}), like `mapreduce(sj -> σ(d, sj), +, array)`.

Reply:
I think you are overengineering this :smile:

Reply:
`reach_blocks!` generate a big MSA "horizontally", e.g. [reach_blocks.jl#L186](https://github.com/JuliaReach/Reachability.jl/blob/master/src/ReachSets/ContinuousPost/BFFPSV18/reach_blocks.jl#L186).

i found that the profiler was giving type instabilities there and began to investigate and put up the results so far in this branch. no need to merge it (i changed the title to WIP to make this clear).